stl/inc/stacktrace

```diff
@@ -300,19 +270,56 @@ public:
     }
 
 private:
-    static constexpr size_t _Max_frames = 0xFFFF;
-
     static constexpr bool _Noex_move = allocator_traits<_Alloc>::propagate_on_container_move_assignment::value
                                     || allocator_traits<_Alloc>::is_always_equal::value;
 
-    struct _Internal_t {
-        explicit _Internal_t() noexcept = default;
-    };
+    static constexpr bool _Noex_swap = allocator_traits<_Alloc>::propagate_on_container_swap::value
+                                    || allocator_traits<_Alloc>::is_always_equal::value;
+
+    static constexpr size_type _Max_frames = 0xFFFF;
+
+    // __declspec(noinline) to make the same behavior for debug and release.
+    // We force the current function to be always noinline and add its frame to skipped.
+
+    _NODISCARD __declspec(noinline) static basic_stacktrace
+        _Current(const unsigned long _Adjusted_skip, const size_type _Max_depth, const allocator_type& _Al) noexcept {
+        // _Adjusted_skip is adjusted outside
+        _TRY_BEGIN
+        basic_stacktrace _Result(_Al);
+
+        // <NOTICE> _Small_frames may be subject to further decreasements to avoid stack burden and
+        // <NOTICE> the cost introduced by the first try should be compensable from uninitialized resize.
+
+        constexpr size_type _Small_frames = 128; // Sufficient in most cases.
+        stacktrace_entry _Buffer[_Small_frames];
+        unsigned short _Caught =
+            __std_stacktrace_capture(_Adjusted_skip, static_cast<unsigned long>((_STD min)(_Max_depth, _Small_frames)),
+                reinterpret_cast<void**>(_Buffer), &_Result._Hash);
+        if (_Caught != _Small_frames) { // actual size
+            _Result._Frames.assign(_Buffer, _Buffer + _Caught);
+            return _Result;
+        }
+
+        size_type _Enough_capacity = (_STD min)(_Max_depth, _Max_frames);
+        // resize(_Enough_capacity), but without value initialization
+        // static_assert(is_implicit_lifetime_v<stacktrace_entry>); TRANSITION, not implemented
+        const auto _Newvec = _Result._Frames._Getal().allocate(_Enough_capacity);
+        _Result._Frames._Change_array(_Newvec, _Enough_capacity, _Enough_capacity);
+
+        _Caught = __std_stacktrace_capture(
+            _Adjusted_skip, static_cast<unsigned long>(_Enough_capacity), _Result._To_voidptr_array(), &_Result._Hash);
+        _Result._Frames.resize(_Caught);
+        return _Result;
 
-    basic_stacktrace(_Internal_t, size_type _Max_depth, const allocator_type& _Al) : _Frames(_Max_depth, _Al) {}
+        _CATCH_ALL
+        return basic_stacktrace{_Al};
+        _CATCH_END
+    }
 
-    _NODISCARD static unsigned long _Adjust_skip(size_t _Skip) noexcept {
-        return _Skip < ULONG_MAX - 1 ? static_cast<unsigned long>(_Skip + 1) : ULONG_MAX;

```

Comment:
For huge values this will return `(unsigned long)ULONG_MAX`, which will become 0 after `+1` in `__std_stacktrace_capture`. Is this intentional behavior?
https://github.com/microsoft/STL/blob/f51733ca5352c712165e46fdbc5d0d395971f0e7/stl/src/stacktrace.cpp#L252